osidb/mixins.py

```diff
@@ -462,3 +462,37 @@ def _validate_acl_identical_to_parent_flaw(self):
 
     class Meta:
         abstract = True
+
+
+class TextChoicesComparisonMixin:
+    """
+    mixin to extend the models.TextChoices classes
+    making them comparable with the standard operators
+    """
+
+    @classmethod
+    def get_choices(cls):
+        """
+        get processed choices
+        """
+        return [choice[0] for choice in cls.choices]
+
+    @property
+    def weight(self):
+        """
+        weight of the instance for the comparison
+        defined by the order of the definition of the choices
+        """
+        return self.get_choices().index(str(self))
+
+    def __lt__(self, other):
+        return self.weight < other.weight
```

Comment:
you need to safeguard against `other` being of a different type, two things can happen here:

1. the other object is of a different enumeration that does implement this mixin, in which case the comparison doesn't make sense
2. the other object does not implement this mixin, in which case this will raise an error because `weight` is not a member of `other`

Reply:
for those reasons and potentially others I would prefer going with something other than overriding the rich comparison operators, e.g. something like `FlawImpact.max(FlawImpact.CRITICAL, FlawImpact.LOW)` but I leave it up to you as I don't know what kind of comparisons will be performed regularly with these enumerations.

Reply:
I agree with the risk of using some other `other` but it seemed to me as to that it would be rather a programmer bug. But no problem to add it. On the other hand I was also thinking about allowing mutual comparison of `FlawImact` and `AffectImpact` (not have a use case but seems meaningful to me) which would make it more complicated to safeguard (the exact class restriction would prevent comparison between these two).

@Elkasitu do you think it could be probably/frequent to compare `FlawImpact` and `AffectImpact` with other classes using the comparison operators so the safeguard would be really needed? To me it seems like not very probably case.

Having it implemented as `FlawImpact.max` would make it syntactically harder to use with larger arrays or generators. Would be doable but I am not fan of this approach.

Reply:
We shouldn't allow comparisons between different enums because that can lead to an apples vs oranges type of error -- in your example it works because both `FlawImpact` and `AffectImpact` have the exact same enum values, but this just means that we should instead have an `Impact` enum that we use in both `Flaw` and `Affect`.

Yes, comparing `ComparableTextChoices` to `other` is a programming error, but the error being raised is not a correct one: it should tell the programmer that he's comparing apples to oranges, not that `attribute x is not a member of y` as that is a confusing error.

See https://docs.python.org/3/library/functools.html#functools.total_ordering for an example of this pattern

```python
def _check_apples_oranges(self, other):
    # type() is necessary to ensure that we're checking that we're comparing
    # apples to apples and oranges to oranges, checking for instances
    # of ComparableTextChoices could lead to such an error
    return type(self) == type(other)

def __eq__(self, other):
    if self._check_apples_oranges(other):
        return self.weight == other.weight
    return NotImplemented
```

Reply:
@Elkasitu thank you for the suggestion! Merging the `FlawImpact` and the `AffectImpact` actually makes a lot of change to me so I have added a commit addressing this. I also extended the `ComparableTextChoices` so it ensures that the compared objects are of the same type. It increases the complexity of `ComparableTextChoices` but should hopefully now be more acceptable. Please have one more look.